packages/core/saved-objects/core-saved-objects-migration-server-internal/src/common/utils/logs.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Logger, LogMeta } from '@kbn/logging';
+import { MigrationLog } from '../../types';
+
+export interface LogAwareState {
+  controlState: string;
+  logs: MigrationLog[];
+}
+
+interface StateTransitionLogMeta extends LogMeta {
+  kibana: {
+    migrations: {
+      state: LogAwareState;
+      duration: number;
+    };
+  };
+}
+
+export const logStateTransition = (
+  logger: Logger,
+  logMessagePrefix: string,
+  prevState: LogAwareState,
```

Comment:
> Does that seems fine to you, or should we go with another folder structure?

Seems good to me. It's not worth getting too tied up in _where_ the code lives, as long as it's somewhat categorized.

Reply:
`src/core` is the remain of old history tbh 😅 . Ideally we would be moving `src/core` to subfolders of `src/common` soon. I just wanted to avoid doing it in the initial PR.